### DISCUSSION TITLE
Java CDS support

### DISCUSSION THREAD

**Discussion by davidxie** [open-telemetry/opentelemetry-java-instrumentation#12005] [Q&A] [answered]
Working on POC to use JDK CDS to speed up startup time for a application with open-telemetry agent 
   -XX:ArchiveClassesAtExit=application.jsa
Because of java agent, need following additional JDK options
  -XX:+UnlockDiagnosticVMOptions -XX:+AllowArchivingWithJavaAgent 

I got following warning:
Java HotSpot(TM) 64-Bit Server VM warning: This archive was created with
AllowArchivingWithJavaAgent. It should be used for testing purposes only and
should not be used in a production environment

Is there any way to use CDS and open-telemetry agent for production environment?

**Comment by laurit** [comment, marked as answer]
Since this message is from the jvm you should turn to your jvm vendor to inquire about the risks involved in using these jvm options and whether they are suitable for production usage.
If using the agent is not an option for spring boot applications you could consider using the [otel spring boot starter](https://opentelemetry.io/docs/zero-code/java/spring-boot-starter/) instead or apply our standalone instrumentation libraries (see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks).
